tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,380 @@
+		testPodWorker1 = createTestPodOnWorkers(cnfWorkerNodeList[1].Definition.Name, portNum8888)
+		testPodList = []*pod.Builder{testPodWorker0, testPodWorker1}
+
+		By("Create a static route to the external Pod network on each worker node")
+		// GetMetalLbVirIP using the metallb virtal IP address variable for test pod IP addresses.
+		ipv4SecurityIPList, err = NetConfig.GetMetalLbVirIP()
+		Expect(err).ToNot(HaveOccurred(), "Failed to retrieve the ipv4SecurityIPList")
+
+		routeMap, err = netenv.BuildRoutesMapWithSpecificRoutes(testPodList, cnfWorkerNodeList, ipv4SecurityIPList)
+		Expect(err).ToNot(HaveOccurred(), "Failed to create route map with specific routes")
+
+		addDeleteStaticRouteOnWorkerNodes(testPodList, routeMap, "add", hubIPv4Network)
+	})
+
+	AfterAll(func() {
+		By("Remove the static route to the external Pod network on each worker node")
+		addDeleteStaticRouteOnWorkerNodes(testPodList, routeMap, "del", hubIPv4Network)
+	})
+
+	Context("custom firewall", func() {
+		AfterEach(func() {
+			By("Define and delete a NFTables custom rule")
+			createMCAndWaitforMCPStable(tsparams.CustomFirewallDelete, mcNftablesName)
+			err := mco.NewMCBuilder(APIClient, mcNftablesName).Delete()
+			Expect(err).ToNot(HaveOccurred(), "Failed to get the machineConfig")
+		})
+
+		It("Verify the creation of a new custom node firewall NFTables table with an ingress rule",
+			reportxml.ID("77412"), func() {
+				By("Setup test environment")
+				masterPod := setupRemoteMultiHopTest(ipv4SecurityIPList, hubIPv4ExternalAddresses,
+					ipv4NodeAddrList, cnfWorkerNodeList, masterNodeList)
+
+				By("Verify ICMP connectivity between the master Pod and the test pods on the workers")
+				err := cmd.ICMPConnectivityCheck(masterPod, ipv4NodeAddrList, interfaceNameNet1)
+				Expect(err).ToNot(HaveOccurred(), "Failed to ping the worker nodes")
+
+				By("Verify ingress TCP traffic over port 8888 between the master Pod and the test pods on the workers")
+
+				err = cmd.ValidateTCPTraffic(masterPod, ipv4NodeAddrList, interfaceNameNet1, frrconfig.ContainerName,
+					portNum8888)
+				Expect(err).ToNot(HaveOccurred(),
+					"Failed to send ingress TCP traffic over port 8888 to the worker nodes")
+
+				By("Verify egress TCP traffic over port 8888 between the test Pod on the worker0 and the master pod")
+				err = cmd.ValidateTCPTraffic(testPodWorker0, []string{masterPodIPv4Address},
+					interfaceNameBrEx, "", portNum8888)
+				Expect(err).ToNot(HaveOccurred(),
+					"Failed to send egress TCP traffic over port 8888 to the pod on the master node")
+
+				By("Verify egress TCP traffic over port 8888 between the test Pod on the worker1 and the master pod")
+				err = cmd.ValidateTCPTraffic(testPodWorker1, []string{masterPodIPv4Address},
+					interfaceNameBrEx, "", portNum8888)
+				Expect(err).ToNot(HaveOccurred(),
+					"Failed to send egress TCP traffic over port 8888 to the pod on the master node")
```

Comment:
Is there any difference between worker0 and worker1? If not, why running the exact same test under the same conditions on worker1?

Reply:
No difference but the custom firewall table is a cluster global configuration so I was checking on both the workers.  As we discussed we dont need to but it also doesnt hurt.